polars/dataframe.ts

```diff
@@ -722,7 +723,7 @@ export interface DataFrame
       rightOn: ValueOrArray<string>;
     } & Omit<JoinOptions, "on">,
   ): DataFrame;
-  join(other: DataFrame, options: { how: "cross"; suffix?: string }): DataFrame;
+  join(other: DataFrame, options: { how: JoinType; suffix?: string }): DataFrame;
```

Comment:
so this signature is to add type hints to the variant `df.join(df2, {how: "cross", ...}) which doesnt take in an `on` or `leftOn + rightOn`. I think we should keep this one as is, but add `how` to the other signature instead. That way we keep the specific type hints for a cross join.